tests/testthat/test-module_teal.R

```diff
@@ -437,13 +437,98 @@ testthat::describe("srv_teal teal_modules", {
     )
   })
 
-  testthat::test_that("receives data with datasets == module$datanames")
+  testthat::test_that("receives data with datasets == module$datanames", {
+    shiny::testServer(
+      app = srv_teal,
+      args = list(
+        id = "test",
+        data = reactive(teal_data(iris = iris, mtcars = mtcars)),
```

Comment:
Do we need to check that the datanames is populated for different data formats we support (`teal_data`, `reactive(teal_data)`, `reactiveVal(teal_data)`, `teal_data_module`)?

Additionally, I think datanames should be checked for three cases:
1. `datanames = "all"` the default case
2. `datanames = c("iris", "mtcars")` When datasets are explicitly mentioned
3. `datanames = c("ADTTE")` When parents should be included using join keys. In this case even `ADSL` should be included if we provide join_keys.

Reply:
@m7pr could you add a test to check the third scenario mentioned by @vedhav

Reply:
Yup, will do @gogonzo

Reply:
https://github.com/insightsengineering/teal/pull/1274/commits/781feeda6e973840314bfa0786a45f3b478d7625